inginious/frontend/templates/task_dispensers/toc.html

```diff
@@ -6,35 +6,38 @@
 {% set level = level|default(3, true) %}
 
 {% for section in sections %}
-    {% if section.is_terminal() %}
-        {% set ns = namespace(section_tasks=[]) %}
-        {% for taskid in section.get_tasks() if taskid in tasks %}
-            {% set ns.section_tasks = ns.section_tasks + [tasks[taskid]] %}
-        {% endfor %}
-        {% with tasks=ns.section_tasks, tag_list=tag_filter_list, list_id=section.get_id(), title=section.get_title(), config=section.get_config() %}
-            {% include "task_dispensers/task_list.html" %}
-        {% endwith %}
-    {% else %}
-        <div id="section_{{ section.get_id() }}" class="section sections-list mb-4">
-            <div class="section_header d-flex justify-content-between divided pr-3" onclick="dropdown(this)">
-                <span class="title h{{level}} mr-3">
-                    {{ _(section.get_title()) }}
-                </span>
-                <span class="divider"></span>
+    {% set ns = namespace(section_tasks=[],have_accessible_tasks=False) %}
+    {% for taskid in section.get_tasks() if taskid in tasks %}
```

Comment:
This will compute the accessibility multiple times. Non-terminal sections `get_tasks()` is recursive and will return the tasks of all the terminal sections it contains.

It would be better on highly structured courses to avoid this behaviour. Maybe it's better to wait for #908 to fix this.

Reply:
I solved multiple computation problem but have to call get_tasks for each section... Probably still improvable?